In [3]:
import pandas as pd
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/googleplaystore.csv"
df = pd.read_csv(url)

duplicados = df.duplicated(subset=['App']).sum()
print("Duplicados encontrados:", duplicados)

df.drop_duplicates(subset=['App'], keep='first', inplace=True)
print("Total de filas tras purgar duplicados:", len(df))

Duplicados encontrados: 1181
Total de filas tras purgar duplicados: 9660


Pregunta A: El scraper cometió errores y guardó la misma app varias veces. Escriban una línea de código para identificar si hay nombres duplicados en la columna App (df.duplicated(subset=['App']).sum()). Si los hay, apliquen .drop_duplicates(subset=['App'], keep='first', inplace=True) para quedarse solo con la primera aparición de cada aplicación. ¿Cuántos registros duplicados purgaron?

Respuesta: Se borraron exactamente 1,181 registros duplicados en la columna 'App, pasando de 10841 a 9660.

In [4]:
df = df[df['Installs'] != 'Free']

df['Installs'] = df['Installs'].str.replace('+', '', regex=False).str.replace(',', '', regex=False).astype(int)

print("Promedio real de descargas:", df['Installs'].mean())

Promedio real de descargas: 7777506.732270421


Pregunta B: La columna Installs tiene texto como "1,000,000+". Un algoritmo no puede procesar eso.
Escriban una instrucción encadenando el método .str.replace() para eliminar el símbolo + y otro para eliminar la coma ,. (Ojo: utilicen regex=False o escapen el símbolo más \+).
Al final de la cadena, utilicen .astype(int) (o float si tienen NaNs) para convertir la columna.
Comprueben el éxito ejecutando df['Installs'].mean(). ¿Cuál es el promedio real de descargas?

Respuesta: El promedio real de descargas es de 7,777,506.73 descargas 

In [ ]:
df['Price'] = df['Price'].str.replace('$', '', regex=False).astype(float)

print(df['Price'].describe())

count    9659.000000
mean        1.099299
std        16.852152
min         0.000000
25%         0.000000
50%         0.000000
75%         0.000000
max       400.000000
Name: Price, dtype: float64


Pregunta C: La columna Price tiene valores como "$4.99" y "0". Apliquen la misma lógica arquitectónica que en la Pregunta B: eliminen el símbolo de dólar $ y conviertan la columna usando .astype(float).

Respuesta: Ahora usa valores flotantes reales

In [ ]:
print("Precio máximo:", df['Price'].max())

print("\nApps con precio mayor a 200 dolares:")
print(df[df['Price'] > 200][['App', 'Price']])

df = df[df['Price'] < 50]
print("\nTotal de registros con Price < 50:", len(df))

df.to_csv('playstore_limpio.csv', index=False)
print("Dataset limpio guardado exitosamente en 'playstore_limpio.csv'.")

Precio máximo: 400.0

Apps con precio mayor a 200 dólares:
                                    App   Price
4197             most expensive app (H)  399.99
4362                         💎 I'm rich  399.99
4367           I'm Rich - Trump Edition  400.00
5351                          I am rich  399.99
5354                     I am Rich Plus  399.99
5355                      I am rich VIP  299.99
5356                  I Am Rich Premium  399.99
5357                I am extremely Rich  379.99
5358                         I am Rich!  399.99
5359                 I am rich(premium)  399.99
5362                      I Am Rich Pro  399.99
5364     I am rich (Most expensive app)  399.99
5366                          I Am Rich  389.99
5369                          I am Rich  399.99
5373                 I AM RICH PRO PLUS  399.99
9917                        Eu Sou Rico  394.99
9934  I'm Rich/Eu sou Rico/أنا غني/我很有錢  399.99

Total de registros con Price < 50: 9636
Dataset limpio guardado exitosamente

Pregunta D: Ahora que el precio es numérico, ejecuten df['Price'].max(). Notarán que la app más cara cuesta alrededor de 400 dólares.
Usen un filtro booleano (df[df['Price'] > 200]) para descubrir cuáles son estas apps. Verán que son aplicaciones basura o de broma.
Si metemos estas apps al modelo, la red neuronal creerá que vender a 400 dólares es un modelo de negocio viable. Sobrescriban el DataFrame conservando únicamente las apps que cuesten menos de 50 dólares (df = df[df['Price'] < 50]).
Guarden su dataset final usando df.to_csv('playstore_limpio.csv', index=False).

Respuesta: Se encontro que el precio mas alto es de 399.99, talvez tiene como un maximo para el costo de una app. Al filtrar se ven 15 apps de broma. Tras filtrar,  el dataset final queda con 9,644 registros.